### ISSUE TITLE
Explore: Remove dead sectors in a network upgrade

### ISSUE THREAD

**Issue by arajasek** [filecoin-project/lotus#11072]
## User Story

I, a Filecoin participant, would like the size of a single state tree (and a single snapshot, etc.) to be smaller. One source of bloat in the state tree is expired SectorInfos. While there are considerations to be made before simply removing them all (eg. would we ever want to allow "reviving" dead sectors?), I would like to know the feasibility and impact of removing all dead sectors in a network upgrade.

## Acceptance criteria

- [ ] Exact amount of state size that would be freed up if we ran this on mainnet today
- [ ] How long such a migration would take on an average machine (both premigration and migration)
- [ ] How many partitions (if any) would be removed in total as a result of this 

## Technical Breakdown

- [ ] Build a migration that runs the CompactPartitions logic for all partitions, on every deadline, for every miner
- [ ] Run this once, and record the change in state size as well as the time taken
- [ ] If worthwhile, re-run with a premigration and other optimizations

**Comment by beck-8** [comment]
Let me understand, if these data are deleted, will this interface be unusable? `StateMinerSectors`
If so, the scope of influence is still very wide, and some tools of lotus itself will rely on it, and personnel related to data analysis will rely on it.

**Comment by arajasek** [comment]
@beck-8 No, that interface should work as before -- it will just return less information, because the "dead" sectors (if any) will no longer be returned.

**Comment by beck-8** [comment]
I see what you mean, I'm talking about this situation. After deleting the data, in most cases will continue in StateMinerActiveSectors , I stand by my opinion, in programming, people who depend on this interface will need some time to adjust.

**Comment by jennijuju** [comment]
exploration ongoing here https://github.com/filecoin-project/FIPs/discussions/795#discussioncomment-6781896

**Comment by rjan90** [comment]
Closing ticket in Lotus. Exploration/any actions with regards to this requires a FIP, and there is a more context information about this exploration in the the [FIP discussion](https://github.com/filecoin-project/FIPs/discussions/795#discussioncomment-6781896).
